File: 프로그래머스/src/프로그래머스_02262022/나누어떨어지는숫자배열.cpp
#include <string>
#include <vector>

using namespace std;
vector<int> answer;
/*algorithm 안사용하기 위해 counting sort 구현...*/
void sort(int max)
{
    vector<bool> sorting(max + 1, false);
    for (auto a : answer)
        sorting[a] = true;
    int cnt = 0;

    for (int i = 0; i < sorting.size(); i++)
    {
        if (sorting[i])
        {
            answer[cnt] = i;
            sorting[i] = false;
            cnt++;
        }
    }
}
vector<int> solution(vector<int> arr, int divisor)
{
    bool none = true;
    int max = -1;
    for (auto a : arr)
    {
        if (a % divisor == 0)
        {
            answer.emplace_back(a);
            none = false;
            max = max > a ? max : a;
        }
    }
    if (none)
        return {-1};
    sort(max);
    return answer;
}

File: 2026/Level1/나누어떨어지는숫자배열.py
# https://school.programmers.co.kr/learn/courses/30/lessons/12910?language=python3
def solution(arr, divisor):
    answer = []
    for x in arr:
        if x % divisor == 0:
            answer.append(x)
    if len(answer) == 0  : 
        answer.append(-1)
    answer.sort()
    return answer